lib/tracker/mode_s/crc_realworld_integration_test.go

```diff
@@ -0,0 +1,256 @@
+package mode_s_test
+
+import (
+	"bufio"
+	"bytes"
+	"errors"
+	"os"
+	"strings"
+	"testing"
+	"time"
+
+	"plane.watch/lib/producer"
+	"plane.watch/lib/tracker/beast"
+	"plane.watch/lib/tracker/mode_s"
+)
+
+// TestCRCWithRealWorldData validates CRC implementation against real Beast format data
+func TestCRCWithRealWorldData(t *testing.T) {
+	// Read the beast file
+	data, err := os.ReadFile("cmd/stress/full-feed.beast")
+	if err != nil {
+		t.Skip("Skipping real-world test: beast file not found")
+		return
+	}
+
+	stats := struct {
+		totalFrames      int
+		validFrames      int
+		invalidCRC       int
+		unknownDF        int
+		parseErrors      int
+		df0, df4, df5    int
+		df11, df16       int
+		df17, df18       int
+		df20, df21, df24 int
+	}{}
+
+	// Use the Beast scanner to properly handle escapes
+	scanner := bufio.NewScanner(bytes.NewReader(data))
+	scanner.Split(producer.ScanBeast)
+
+	for scanner.Scan() {
+		beastFrame := scanner.Bytes()
+
+		// Parse Beast frame
+		frame, err := beast.NewFrame(beastFrame, false)
+		if err != nil {
+			continue // Skip bad beast frames
+		}
+
+		// Decode the Beast frame first to ensure Mode S frame is initialized
+		err = frame.Decode()
+		if err != nil {
+			if errors.Is(err, mode_s.ErrNoOp) {
+				continue // Skip NoOp frames
+			}
+			// Some other decode error - skip
+			continue
```

Comment:
what other errors are you expecting to find here?

skipping them seems like it could cover up future decoding bugs

Reply:
I've added debug logging for any frames not captured.